beacon-chain/blockchain/chain_info.go

```diff
@@ -338,7 +337,7 @@ func (s *Service) IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool,
 	if err == nil {
 		return optimistic, nil
 	}
-	if err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
+	if err != doublylinkedtree.ErrNilNode {
```

Comment:
errors.Is is the preferable approach to checking errors in this manner.

https://pkg.go.dev/errors#pkg-overview


```suggestion
	if !errors.Is(err, doublylinkedtree.ErrNilNode) {
```